mypy/checkexpr.py

```diff
@@ -1439,6 +1439,9 @@ def visit_index_expr_helper(self, e: IndexExpr) -> Type:
                 return AnyType()
         elif isinstance(left_type, TypedDictType):
             return self.visit_typeddict_index_expr(left_type, e.index)
+        elif (isinstance(left_type, CallableType)
+              and left_type.is_type_obj() and left_type.type_object()):
```

Comment:
It looks like this already works correctly, but I think it would be safer to check for ``left_type.type_object().is_enum``